generic/build

```diff
@@ -252,10 +252,10 @@ build_dep() {
 }
 
 build_openvpn() {
-
 	echo "Build openvpn"
 	cd "${BUILDROOT}/openvpn-${OPENVPN_VERSION}" || die "cd openvpn"
 	autoreconf -ivf # this allows to specify github url as OPENVPN_URL
+	CFLAGS="${CFLAGS} -O2 -flto"
```

Comment:
The build scripts supposedly supports several platforms and compilers, so hard-coding -O2 here may not be the right approach. Though I've no idea whether anyone uses this for anything other than mingw builds.

Ideally -g -O2 should be added by `configure` which it will do if we undef CFLAGS. Not an option for Windows builds as we want to add --Wl,dynamicbase etc..

I would suggest to add "-O2 -flto" to EXTRA_TARGET_CFLAGS in build.vars or to TARGET_CFLAGS in this file (around line 350)  over-ridden by  --configuration specific values in the case statement around line 410.

Reply:
> I would suggest to add "-O2 -flto" to EXTRA_TARGET_CFLAGS in build.vars or to TARGET_CFLAGS in this file (around line 350) over-ridden by --configuration specific values in the case statement around line 410.

This will modify `CFLAGS` value for all builds (openvpn and dependencies), including, for example openssl. This conflicts with openssl's own optimization flags, see both `O3` and `O2` below:

`i686-w64-mingw32-gcc  -I. -Icrypto/include -Iinclude -DOPENSSL_USE_APPLINK -DDSO_WIN32 -DNDEBUG -DOPENSSL_THREADS -DOPENSSL_NO_STATIC_ENGINE -DOPENSSL_PIC -DOPENSSL_BN_ASM_PART_WORDS -DOPENSSL_IA32_SSE2 -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_GF2m -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DRC4_ASM -DMD5_ASM -DRMD160_ASM -DAES_ASM -DVPAES_ASM -DWHIRLPOOL_ASM -DGHASH_ASM -DECP_NISTZ256_ASM -DPADLOCK_ASM -DPOLY1305_ASM -DOPENSSLDIR="\"/etc/ssl\"" -DENGINESDIR="\"/lib/engines-1_1\"" -DL_ENDIAN -DWIN32_LEAN_AND_MEAN -DUNICODE -D_UNICODE -m32 -Wall -O3 -fomit-frame-pointer -D_MT  -O2 -static-libgcc -D_WINDLL -MMD -MF crypto/bf/bf_ofb64.d.tmp -MT crypto/bf/bf_ofb64.o -c -o crypto/bf/bf_ofb64.o crypto/bf/bf_ofb64.c`

Reply:
Then we may need something like OPENVPN_CFLAGS_OPT defined based on the compiler/platform and use it in build_openvpn(). Or something similar that doesn't break the build script on any of the supported platforms.

Reply:
Added `OPENVPN_CFLAGS_OPT` define, which is set in `windows-nsis\build-complete` - it is only used to build windows installers.